### PR TITLE
add rssi to discovered device

### PR DESCRIPTION
**Pull request by remonh87**

### CODE REVIEW

File: lib/src/model/discovered_device.dart [hunk 1]
```diff
@@ -27,7 +27,14 @@ class DiscoveredDevice extends $DiscoveredDevice {
...
 
-  const DiscoveredDevice({@required this.id, @required this.name, @required this.serviceData});
+  @CustomEquality(Ignore())
+  final int rssi;
+
+  const DiscoveredDevice(
+      {@required this.id,
+      @required this.name,
+      @required this.serviceData,
+      @required this.rssi});
```
Comment (keke-dandois) [inline lib/src/model/discovered_device.dart]: add a ,

File: lib/src/converter/protobuf_converter.dart [hunk 2]
```diff
@@ -31,7 +31,10 @@ class ProtobufConverter {
     return ScanResult(
       result: resultFrom(
           getValue: () => DiscoveredDevice(
-              id: message.id, name: message.name, serviceData: serviceData),
+              id: message.id,
+              name: message.name,
+              serviceData: serviceData,
+              rssi: message.rssi),
```
Comment (keke-dandois) [inline lib/src/converter/protobuf_converter.dart]: add a ,

File: lib/src/model/discovered_device.dart [hunk 3]
```diff
@@ -27,7 +27,14 @@ class DiscoveredDevice extends $DiscoveredDevice {
   @CustomEquality(DeepCollectionEquality())
   final Map<Uuid, Uint8List> serviceData;
 
-  const DiscoveredDevice({@required this.id, @required this.name, @required this.serviceData});
+  @CustomEquality(Ignore())
```
Comment (miguelcmedeiros) [inline lib/src/model/discovered_device.dart]: Why is this field ignored for equality?

Comment (remonh87) [reply lib/src/model/discovered_device.dart]: My reasoning behind it is that a device with the same id, service data and name but with a different RSSI would be seen as a different object and I think this shouldn't be the case.

**Pull request merged**